src/go/rpk/pkg/config/schema.go

```diff
@@ -142,8 +142,8 @@ type RpkKafkaApi struct {
 }
 
 type RpkAdminApi struct {
-	Addresses []SocketAddress `yaml:"addresses,omitempty" mapstructure:"addresses,omitempty" json:"addresses"`
-	TLS       *TLS            `yaml:"tls,omitempty" mapstructure:"tls,omitempty" json:"tls"`
+	Addresses []string `yaml:"addresses,omitempty" mapstructure:"addresses,omitempty" json:"addresses"`
```

Comment:
Will this break existing configurations? Do we expect the admin api to be widely used?

If so we _could_ be backwards compatible in a kludgy way bay having `Addresses` be an interface{}... and then it gets worse from there: we would type switch on `map[string]json.RawMessage` and `[]string`, and if the former, unmarshal each value into a ScoketAddress.

Reply:
If someone is currently using the "NamedSocket" structure, then yes, it will break. However, I don't think anyone is because this was released last week and [the docs](https://vectorized.io/docs/configuration) match the correct format.

Reply:
Cool, good.